### PR TITLE
[GCS]Fix ServiceBasedGcsClientTest bug

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
BUG:
image
In the test case, we test actor/task/object table unsubscribe function. We will send messages after unsubscribe and expect that it will not receive new subscription messages, but in fact, it will still receive new subscription messages because unsubscribe is asynchronous operation. When we send new messages, unsubscribe maybe not completed.
Approach:
After unsubscribe is executed, we judge whether unsubscribe is successfully executed.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
@ffbin Can you improve description a little bit better?

1. More high level description why the bug occured.
2. What's your approach.

**Comment by stephanie-wang** [comment]
> @ffbin Can you improve description a little bit?
> 
>     1. More high level description why the bug occured.
> 
>     2. What's your approach.

+1. Please make sure to do this *before* you ask for a review, as it takes time and effort on the part of the reviewer to understand the PR without this. I will review once these are added.

**Comment by ffbin** [comment]
@rkooo567 @stephanie-wang I have update the description, thanks :)

File: src/ray/gcs/pubsub/gcs_pub_sub.cc [hunk 1]
```diff
@@ -174,5 +174,12 @@ std::string GcsPubSub::GenChannelPattern(const std::string &channel,
   return pattern.str();
 }
 
+bool GcsPubSub::IsUnsubscribed(const std::string &channel, const std::string &id) {
+  std::string pattern = GenChannelPattern(channel, id);
+
+  absl::MutexLock lock(&mutex_);
```
Comment (rkooo567) [inline src/ray/gcs/pubsub/gcs_pub_sub.cc:180]: Not related to the PR, but why do we need mutex here? I thought gcs server was single threaded?

**Review by rkooo567 — APPROVED** [review]

File: src/ray/gcs/pubsub/gcs_pub_sub.cc (hunk 1 above)
Comment (ffbin) [reply src/ray/gcs/pubsub/gcs_pub_sub.cc:180]: This is the GcsPubSub called by the gcs client. The GcsPubSub will have a separate thread(client_io_service_thread_) to process the unsubscribe reply, and we will query whether unsubscribe is successful in the UT main thread, so there will be multi threading problems.

**Comment by rkooo567** [comment]
@ffbin Please resolve the merge conflict! And in the master, the Windows CI is supposed to pass now (we blacklisted all failing tests)!

**Pull request merged**